api/src/services/collections.ts

```diff
@@ -53,127 +53,131 @@ export class CollectionsService {
+								collection: payload.collection!,
+							};
+						}
 
-					return field;
-				});
+						return field;
+					});
 
-				const fieldsService = new FieldsService({ knex: trx, schema: this.schema });
+					const fieldsService = new FieldsService({ knex: trx, schema: this.schema });
 
-				await trx.schema.createTable(payload.collection, (table) => {
-					for (const field of payload.fields!) {
-						if (field.type && ALIAS_TYPES.includes(field.type) === false) {
-							fieldsService.addColumnToTable(table, field);
+					await trx.schema.createTable(payload.collection, (table) => {
+						for (const field of payload.fields!) {
+							if (field.type && ALIAS_TYPES.includes(field.type) === false) {
+								fieldsService.addColumnToTable(table, field);
+							}
 						}
-					}
-				});
+					});
 
-				const fieldItemsService = new ItemsService('directus_fields', {
-					knex: trx,
-					accountability: this.accountability,
-					schema: this.schema,
-				});
+					const fieldItemsService = new ItemsService('directus_fields', {
+						knex: trx,
+						accountability: this.accountability,
+						schema: this.schema,
+					});
 
-				const fieldPayloads = payload.fields!.filter((field) => field.meta).map((field) => field.meta) as FieldMeta[];
-				await fieldItemsService.createMany(fieldPayloads);
-			}
+					const fieldPayloads = payload.fields!.filter((field) => field.meta).map((field) => field.meta) as FieldMeta[];
+					await fieldItemsService.createMany(fieldPayloads);
+				}
 
-			if (payload.meta) {
-				const collectionItemsService = new ItemsService('directus_collections', {
-					knex: trx,
-					accountability: this.accountability,
-					schema: this.schema,
-				});
+				if (payload.meta) {
+					const collectionItemsService = new ItemsService('directus_collections', {
+						knex: trx,
+						accountability: this.accountability,
+						schema: this.schema,
+					});
+
+					await collectionItemsService.createOne({
+						...payload.meta,
+						collection: payload.collection,
+					});
+				}
 
-				await collectionItemsService.createOne({
-					...payload.meta,
-					collection: payload.collection,
-				});
+				return payload.collection;
+			});
+		} finally {
+			if (this.cache && env.CACHE_AUTO_PURGE && opts?.autoPurgeCache !== false) {
+				await this.cache.clear();
 			}
 
-			return payload.collection;
-		});
-
-		if (this.cache && env.CACHE_AUTO_PURGE && opts?.autoPurgeCache !== false) {
-			await this.cache.clear();
+			await clearSystemCache();
 		}
 
-		await clearSystemCache();
-
 		return payload.collection;
 	}
 
 	/**
 	 * Create multiple new collections
 	 */
 	async createMany(payloads: RawCollection[], opts?: MutationOptions): Promise<string[]> {
-		const collections = await this.knex.transaction(async (trx) => {
-			const service = new CollectionsService({
-				schema: this.schema,
-				accountability: this.accountability,
-				knex: trx,
-			});
-
-			const collectionNames: string[] = [];
+		try {
```

Comment:
Since we are calling `this.createOne` and since this already does the same thing, maybe we can run into problems here so, imo we may not use this logic here 🤔